internal/kafka/internal/handlers/admin_kafka_test.go

```diff
@@ -244,40 +246,53 @@ func Test_Update(t *testing.T) {
 		{
 			name: "should return an error if retrieving kafka to update fails",
 			fields: fields{
-				kafkaService: &services.KafkaServiceMock{
-					GetFunc: func(ctx context.Context, id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
-						return nil, errors.GeneralError("test")
+				clusterService: &services.ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return nil, nil
+					},
+					IsStrimziKafkaVersionAvailableInClusterFunc: func(cluster *api.Cluster, strimziVersion, kafkaVersion, ibpVersion string) (bool, error) {
+						return true, nil
+					},
+					CheckStrimziVersionReadyFunc: func(cluster *api.Cluster, strimziVersion string) (bool, error) {
+						return true, nil
 					},
 				},
-			},
-			args: args{
-				url:  "/kafkas/{id}",
-				body: []byte(`{}`),
-			},
-			wantStatusCode: http.StatusInternalServerError,
-		},
-		{
-			name: "should return an error if kafka to update can't be found",
-			fields: fields{
 				kafkaService: &services.KafkaServiceMock{
 					GetFunc: func(ctx context.Context, id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
-						return nil, nil
+						return &dbapi.KafkaRequest{
+							Status: constants.KafkaRequestStatusPreparing.String(),
+							Meta: api.Meta{
+								ID: "id",
+							},
+							ClusterID:              "cluster-id",
+							ActualKafkaIBPVersion:  "2.7",
+							DesiredKafkaIBPVersion: "2.7",
+							ActualKafkaVersion:     "2.7",
+							DesiredKafkaVersion:    "2.7",
+							DesiredStrimziVersion:  "2.7",
+							KafkaStorageSize:       "100",
+						}, errors.GeneralError("test")
```

Comment:
Why returning both a return value and an error? Can't we just return `nil, errors.GeneralError("test")`?

Before this wasn't returning any error on `kafka no found` (was just returning `nil`): is the behaviour changed?

Reply:
Thanks for pointing this out :) no, it was just the case that I copied the kafka request from one of my other tests. I'll switch it out in the morning.

Reply:
Thanks @ziccardi, this has been resolved in the latest push :)